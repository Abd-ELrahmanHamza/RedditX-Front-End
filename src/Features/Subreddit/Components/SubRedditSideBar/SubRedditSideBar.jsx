import CommunityCard from "../CommunityCard/CommunityCard";
import ModeratorWidget from "../ModeratorWidget/ModeratorWidget";
import RulesWidget from "../RulesWidget/RulesWidget";
import { Container, InnerContainer } from "./SubRedditSideBar.styled";
const SubRedditSideBar = () => {
  return (
    <Container>
      <InnerContainer>
        <CommunityCard />
        {true && <RulesWidget />}
        <ModeratorWidget />
      </InnerContainer>
    </Container>
  );
};

export default SubRedditSideBar;
