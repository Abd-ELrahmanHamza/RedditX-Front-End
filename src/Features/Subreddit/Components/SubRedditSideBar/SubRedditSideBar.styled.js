import styled from "styled-components";

export const Container = styled.div`
  box-sizing: border-box;
  margin-left: 24px;
  margin-top: 0;
  flex: 0 0 312px;
  width: 312px;

  @media (max-width: 960px) {
    display: none;
  }
`;

export const InnerContainer = styled.div`
  width: inherit;
  display: flex;
  flex-direction: column;
  height: 100%;
`;
