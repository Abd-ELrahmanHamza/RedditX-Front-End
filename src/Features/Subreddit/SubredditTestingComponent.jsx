import { ThemeProvider } from "styled-components";
import defaultTheme from "Theme/defaultTheme";
import lightTheme from "Theme/lightTheme";
import { BrowserRouter as Router } from "react-router-dom";
import { SubRedditProvider } from "Features/Subreddit/Contexts/SubRedditProvider";
import { SubRedditIDProvider } from "./Contexts/SubRedditIDProvider";
import { IsModeratorProvider } from "Features/Subreddit/Contexts/IsModeratorProvider";
import { IsBannedProvider } from "Features/Subreddit/Contexts/IsBannedProvider";
import { IsMutedProvider } from "Features/Subreddit/Contexts/IsMutedProvider";

const theme = { ...defaultTheme, ...lightTheme };

/**
 * Function that wraps Children props inside a Browser Router and a Theme Provider
 * @returns {React.Component}The component is being returned.
 */
const SubredditTestingComponent = ({ children }) => {
  return (
    <Router>
      <ThemeProvider theme={theme}>
        <SubRedditProvider>
          <SubRedditIDProvider>
            <IsModeratorProvider>
              <IsBannedProvider>
                <IsMutedProvider>{children}</IsMutedProvider>
              </IsBannedProvider>
            </IsModeratorProvider>
          </SubRedditIDProvider>
        </SubRedditProvider>
      </ThemeProvider>
    </Router>
  );
};

export default SubredditTestingComponent;
