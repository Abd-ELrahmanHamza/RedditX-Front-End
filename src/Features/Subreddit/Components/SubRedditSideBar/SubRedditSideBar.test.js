import { render } from "@testing-library/react";
// import Enzyme, { shallow } from "enzyme";
import SubRedditSideBar from "./SubRedditSideBar";
import SubredditTestingComponent from "Features/Subreddit/SubredditTestingComponent";

// Enzyme.configure({ adapter: new Adapter() });

describe("sidebar of subreddit", () => {
  // it("should render without crashing", () => {
  //   expect(shallow(<SubRedditSideBar />)).toMatchSnapshot();
  // });

  it("should render sidebar without crashing", () => {
    render(
      <SubredditTestingComponent>
        <SubRedditSideBar />
      </SubredditTestingComponent>
    );
  });
});
