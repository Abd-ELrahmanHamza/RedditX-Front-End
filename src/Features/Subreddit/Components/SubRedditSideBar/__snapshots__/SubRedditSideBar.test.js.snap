// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`sidebar of subreddit should render without crashing 1`] = `ShallowWrapper {}`;
